crates/criticaltrust/src/revocation_info.rs

```diff
@@ -0,0 +1,38 @@
+use crate::keys::KeyRole;
+use crate::signatures::Signable;
+use crate::NoRevocationsCheck;
+use serde::{Deserialize, Serialize};
+use time::macros::datetime;
+use time::OffsetDateTime;
+
+/// Holds hashes of revoked content which are included as a part of the [`KeysManifest`].
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct RevocationInfo {
+    pub revoked_content_sha256: Vec<String>,
+    #[serde(with = "time::serde::rfc3339")]
+    pub expires_at: OffsetDateTime,
+}
+
+impl RevocationInfo {
+    pub fn new() -> Self {
+        RevocationInfo {
+            revoked_content_sha256: Vec::new(),
```

Comment:
Why initialize this as empty? That is not a valid sha, `RevocationInfo::new(...)` should produce a valid, usable thing. We should avoid allowing invalid states from being represented at all.

Reply:
The list for revoked content that comes from the server also is empty. So I assumed that this is represented anyway in the data.

Reply:
I feel like the assumption that the revoked content list is always empty may be mistaken.

In some tests (and in real world) sometimes that list will not be empty. As this API currently stands it is assumed the developer knows they must then manipulate the `recoved_content_sha256` field to add content.

Reply:
Yea! Makes sense. I updated the constructor (new) to accept list too.